lib/codegen/recorder.dart

```diff
@@ -257,7 +257,8 @@ class Recorder {
     // [node] is the initialization expression, we walk up to get to the actual
     // member declaration where the metadata is attached to.
     while (node is! ClassMember) node = node.parent;
-    return node.metadata.map(_convertAnnotation).toList();
+    return (node as ClassMember).metadata.map(_convertAnnotation).toList() as
```

Comment:
It would be great if map could also take a generic arg as well, not sure who to talk to about this though. @sigmundch do you know what this would take (and if its feasible?).